docs/docs/understanding/workflows/basic_flow.md

```diff
@@ -0,0 +1,174 @@
+# Basic workflow
+
+## Getting started
+
+Workflows are built into LlamaIndex core, so to use them all you need is
+
+```
+pip install llama-index-core
+```
+
+During development you will probably find it helpful to visualize your workflow; you can use our built-in visualizer for this by installing it:
+
+```
+pip install llama-index-utils-workflow
+```
+
+## Dependencies
+
+The minimal dependencies for a workflow are:
+
+```python
+from llama_index.core.workflow import (
+    StartEvent,
+    StopEvent,
+    Workflow,
+    step,
+)
+```
+
+## Single-step workflow
```

Comment:
The type annotations are actually crucial to
- validating the workflow statically before runtime
- deciding which steps accept which events

We might want to explclity call this out, unless I missed this elsewhere

Reply:
Good call!